libexec/python/docker/api.py

```diff
@@ -51,13 +49,34 @@ def create_runscript(cmd,base_dir):
     output_file = write_file(runscript,content)
     return output_file
 
-def create_envfile(env, base_dir):
-    envfile = "%s/environment" % base_dir
+
+def create_envfile(base_dir,env=None,manifest=None):
+    '''create_envfile will generate a file called environment
+    in the base_dir of an image
+    :param base_dir: the base_dir to write the file to
+    :param manifest: the manifest to generate the Env for.
+    :param env: a list of environment variables to export, if manifest
+    is not provided
+    '''
+    if env == None:
+        if manifest == None:
+            logger.error("env or manifest must be provided, exiting")
+            sys.exit(1)    
+        env = get_config(manifest, 'Env').split("\n")
```

Comment:
It seems a little weird to me to have all these optional arguments everywhere, it makes the interface of the function more difficult to understand IMO. I have troubles understanding how they will be used.
Do we really need to make `env` optional? The file is `docker/api.py`, should it allow setting a custom environment unrelated to Docker?
`base_dir` + mandatory `manifest` would be enough.